### PR TITLE
!str #19710 Use Java types in Attributes Java API

### PR DESCRIPTION
**Pull request by johanandren**
Refs #19710

### CODE REVIEW

**Comment by drewhk** [comment]
LGTM

**Comment by johanandren** [comment]
Meh. There was nothing wrong with that commit.

**Comment by johanandren** [comment]
PLS BUILD

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala [hunk 1]
```diff
@@ -60,6 +60,17 @@ class AttributesSpec extends AkkaSpec with ScalaFutures {
...
     }
+
+    val attributes = Attributes.name("a") and Attributes.name("b") and Attributes.inputBuffer(1, 2)
+
+    "give access to first attribute" in {
+      attributes.getFirst[Name] should ===(Some(new Attributes.Name("a")))
+    }
+
+    "give access to attribute byt type" in {
+      attributes.get[Name] should ===(Some(new Attributes.Name("b")))
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala]: Name does not have an apply method?

**Comment by ktoso** [comment]
LGTM

**Comment by ktoso** [comment]
```
[info] Compiling 108 Scala sources and 1 Java source to /localhome/jenkinsakka/workspace/pr-validator-per-commit-jenkins/akka-stream/target/classes...
[error] /localhome/jenkinsakka/workspace/pr-validator-per-commit-jenkins/akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:448: missing parameter type for expanded function ((x$2) => x$2.decider)
[error]     val decider = inheritedAttributes.getAttribute(classOf[SupervisionStrategy]).map(_.decider).getOrElse(Supervision.stoppingDecider)
[error]                                                                                      ^
[error] one error found
```

**Comment by johanandren** [comment]
Needs a rebase, that's why it failed.

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala [hunk 2]
```diff
@@ -60,6 +60,17 @@ class AttributesSpec extends AkkaSpec with ScalaFutures {
         attributes.get[Name] should contain(Name("attributesSink"))
       }
     }
+
+    val attributes = Attributes.name("a") and Attributes.name("b") and Attributes.inputBuffer(1, 2)
```
Comment (viktorklang) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala]: Seeing `Name("attributesSink")` above I think you may be able to drop the `Attributes.` part.

File: akka-stream/src/main/scala/akka/stream/Attributes.scala [hunk 3]
```diff
@@ -48,32 +51,30 @@ final case class Attributes(attributeList: List[Attributes.Attribute] = Nil) {
...
+      (acc, attr) ⇒ if (c.isInstance(attr)) c.cast(attr) else acc)
+    )
 
   /**
    * Java API: Get the first (least specific) attribute of a given `Class` or subclass thereof.
    */
-  def getFirstAttribute[T <: Attribute](c: Class[T]): Option[T] =
-    attributeList.find(c isInstance _).map(c cast _)
+  def getFirstAttribute[T <: Attribute](c: Class[T]): Optional[T] =
+    attributeList.find(c isInstance _).map(c cast _).asJava
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/Attributes.scala]: collectFirst?

File: akka-stream/src/main/scala/akka/stream/Attributes.scala [hunk 4]
```diff
@@ -102,14 +103,18 @@ final case class Attributes(attributeList: List[Attributes.Attribute] = Nil) {
...
+  }
 
   /**
    * Scala API: Get the first (least specific) attribute of a given type parameter T `Class` or subclass thereof.
    */
-  def getFirst[T <: Attribute: ClassTag]: Option[T] =
-    getFirstAttribute(classTag[T].runtimeClass.asInstanceOf[Class[T]])
+  def getFirst[T <: Attribute: ClassTag]: Option[T] = {
+    val c = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+    attributeList.find(c isInstance _).map(c cast _)
```
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/Attributes.scala]: collectFirst?

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala (hunk 2 above)
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/AttributesSpec.scala]: `Name` is an `Attribute` we want `Attributes` here.

**Comment by johanandren** [comment]
PLS BUILD

**Pull request merged**